presto-main/src/main/java/com/facebook/presto/operator/ScanFilterAndProjectOperator.java

```diff
@@ -235,6 +235,7 @@ public Page getOutput()
             }
             else {
                 pageSource = source;
+                operatorContext.updateStats(pageSource.getRuntimeStats());
```

Comment:
Initially, I thought this wasn't needed as well, but this is required to handle an edge case. Specifically, if the page source doesn't return at least 1 page, then the runtimestats is never recorded. In our case, this can happen if all blocks were skipped.

Reply:
I see. Is there a way to detect this case and only do the extra recording for this special case instead of doing it for all cases?

Reply:
recordInputStats is called in recordProcessedInput when page != null. Does it work if we move the recordInputStats out of the if page != null branch and always record it?

Reply:
Sorry about the last response. I tested this locally and yes, this should work. I've moved the recordInputStats() outside the page != null check for ScanFilterAndProject and TableNode.